src/OpenTelemetry/Metrics/Tags.cs

```diff
@@ -0,0 +1,107 @@
+// <copyright file="Tags.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+
+namespace OpenTelemetry.Metrics
+{
+    internal readonly struct Tags : IEquatable<Tags>
+    {
+        public Tags(string[] keys, object[] values)
+        {
+            this.Keys = keys;
+            this.Values = values;
+        }
+
+        public readonly string[] Keys { get; }
+
+        public readonly object[] Values { get; }
+
+        public static bool operator ==(Tags tag1, Tags tag2) => tag1.Equals(tag2);
+
+        public static bool operator !=(Tags tag1, Tags tag2) => !tag1.Equals(tag2);
+
+        public readonly override bool Equals(object obj)
+        {
+            if (!(obj is Tags))
```

Comment:
nit: I forget what C# version we have but on newer, you can make this a bit simpler:

```csharp
 return obj is not Tags other
   ? false
   : this.Equals(other);
```

Something like that 😄

Reply:
Simplified this.